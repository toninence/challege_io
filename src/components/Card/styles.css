.carta-box {
  width: 250px;
  height: 250px;
  position: relative;
  perspective: 1000px;
}

.carta-box:hover .carta {
  transform: rotateY(180deg);
}

.carta {
  transform-style: preserve-3d;
  transition: all 0.4s linear;
  height: 100%;
}

.cara {
  position: absolute;
  backface-visibility: hidden;
}

.cara.detras {
  transform: rotateY(180deg);
}

.navlink {
  color: rgba(138, 123, 123, 1);
}

.navlink:visited {
  color: rgba(138, 123, 123, 1);
}


@media screen and (width < 1180px) {
  .carta-box:hover .carta {
    transform: none;
  }
  .cara {
    position: relative;
    backface-visibility: visible;
  }
  .cara.detras{
    transform: rotateY(0deg);
    height: 300px;
  }
  .carta-box {
    height: auto;
  }
}