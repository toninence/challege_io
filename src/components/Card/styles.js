import styled from "styled-components";

//carta-box
export const CardContainer = styled.div`
  margin: 8px 10px;
  box-shadow: 5px 5px 5px #111;
  font-size: .9em;
`;
//carta
export const CardComponent = styled.div`
  color: ${({theme}) => theme.colors.primary};
  border-radius: 5px;
`;
//cara
export const ImageContainer = styled.div`
  width: 100%;
`;

//cara detras
export const TextContainer = styled.div`
  padding: 5px 15px;
  border-radius: 5px;
  width: 100%;
  height: 100%;
  overflow: scroll;
  display: flex;
  flex-flow: column;
  justify-content: space-evenly;
  color: ${({theme})=> theme.colors.text};
  background-color: ${({theme}) => theme.colors.bgPrimary};
`;

export const Image = styled.img`
  object-fit: contain;
  width: 100%;
  height: auto;
  border-radius: 5px;

  @supports (object-fit: contain) {
    width: 100%;
    object-fit: contain;
    object-position: top left;
  }
`;

export const Name = styled.h3`
  font-weight: bold;
`;

export const Status = styled.p`
  display: flex;
  align-items: center;
`;
export const Species = styled.span``;
export const Origin = styled.p`
  display: inline-block;
  font-weight: bold;
`;
export const Location = styled.p`
  display: inline-block;
  font-weight: bold;
  :visited {
    color: red;
  }
`;
export const Button = styled.button`
  width: 100%;
  border: none;
  background-color: red;
  height: 25px;
`;

