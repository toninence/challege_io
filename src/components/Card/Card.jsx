import { useContext, useEffect, useRef, useState } from "react";
import {
  Button,
  CardComponent,
  CardContainer,
  Image,
  ImageContainer,
  Location,
  Name,
  Origin,
  Species,
  Status,
  TextContainer,
} from "./styles";
import Favorites from "../Favorites";
import StatusIcon from "../StatusIcon";
import "./styles.css";
import { NavLink } from "react-router-dom";
import { FavoritesContext } from "../../context/Favorites/FavoritesContext";
// import { FavoritesContext } from "../../context/Favorites/FavoritesContext";

export const Card = (props) => {

  const {character, toggleFavorites, checkFavorite} = props;
  const { name, image, species, status, url, origin, location, id } = character;
  const { state } = useContext(FavoritesContext);
  let locationNumber = location.url.split("/");
  locationNumber = locationNumber[locationNumber.length - 1];

  const isFavorite = checkFavorite(character);

  return (
    <CardContainer className="carta-box">
      <CardComponent className="carta">
        <ImageContainer className="cara">
          <Image src={image} alt={`image of ${name}`} />
        </ImageContainer>
        <TextContainer className="cara detras">
          <Name>{name}</Name>
          <Status>
            <StatusIcon status={status} />
            {status} - {species}
          </Status>
          <div>
            <span>From: </span>
            <Origin> {origin.name}</Origin>
          </div>
          <div>
            <span>Last known location: </span>
            <NavLink className="navlink" to={`/location/${locationNumber}`}>
              <Location>{location.name}</Location>
            </NavLink>
          </div>
          <div style={{ display: "flex", justifyContent: "space-between" }}>
            <Favorites
              isFavorite={isFavorite}
              toggleFavorites={toggleFavorites}
              character={character}
            />
            <NavLink to={`/character/${id}`}>{/* <PlusButton /> */}</NavLink>
          </div>
        </TextContainer>
      </CardComponent>
    </CardContainer>
  );
};
