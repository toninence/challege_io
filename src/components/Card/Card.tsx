import { useContext } from "react";
import Favorites from "../Favorites";
import StatusIcon from "../StatusIcon";
import "./styles.css";
import { NavLink } from "react-router-dom";
import { FavoritesContext } from "../../context/Favorites/FavoritesContext";
import { CharacterProps } from '../../interfaces/interfaces';
// import { FavoritesContext } from "../../context/Favorites/FavoritesContext";
const {
  Button,
  CardComponent,
  CardContainer,
  Image,
  ImageContainer,
  Location,
  Name,
  Origin,
  Species,
  Status,
  TextContainer,
} = require("./styles");

interface CardProps {
  character: {
    name: string,
    image: string,
    species: string,
    status: string,
    url: string,
    origin: {name:string},
    location: {name:string, url:string},
    id: number
  };
  toggleFavorites: (payload:CharacterProps)=>void;
  checkFavorite: (character:CharacterProps) => Boolean;
};

export const Card = ({character, toggleFavorites, checkFavorite}:CardProps) => {
  // const { character, toggleFavorites, checkFavorite } = props;
  const { name, image, species, status, url, origin, location, id } = character;
  const { state } = useContext(FavoritesContext);
  let locationNumber = location.url.split("/");
  let lNumber = locationNumber[locationNumber.length - 1];

  const isFavorite = checkFavorite(character);

  return (
    <CardContainer className="carta-box">
      <CardComponent className="carta">
        <ImageContainer className="cara">
          <Image src={image} alt={`image of ${name}`} />
        </ImageContainer>
        <TextContainer className="cara detras">
          <Name>{name}</Name>
          <Status>
            <StatusIcon status={status} />
            {status} - {species}
          </Status>
          <div>
            <span>From: </span>
            <Origin> {origin.name}</Origin>
          </div>
          <div>
            <span>Last known location: </span>
            <NavLink className="navlink" to={`/location/${lNumber}`}>
              <Location>{location.name}</Location>
            </NavLink>
          </div>
          <div style={{ display: "flex", justifyContent: "space-between" }}>
            <Favorites
              isFavorite={isFavorite}
              toggleFavorites={toggleFavorites}
              character={character}
            />
            <NavLink to={`/character/${id}`}>{/* <PlusButton /> */}</NavLink>
          </div>
        </TextContainer>
      </CardComponent>
    </CardContainer>
  );
};
