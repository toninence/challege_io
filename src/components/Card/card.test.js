import { fireEvent, prettyDOM, render } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import { Card } from "./Card";
import { ThemeProvider } from "styled-components";
import { BrowserRouter as Router, Route } from "react-router-dom";

const character = {
  id: 1,
  name: "Rick Sanchez",
  image: "https://rickandmortyapi.com/api/character/avatar/1.jpeg",
  species: "Human",
  status: "Alive",
  url: "url",
  origin: {
    name: "Earth (C-137)",
    url: "https://rickandmortyapi.com/api/location/1",
  },
  location: {
    name: "Citadel of Ricks",
    url: "https://rickandmortyapi.com/api/location/3",
  },
};
const themes = {
  light: {
    colors: {
      primary: "#000",
      text: "#000",
      bgPrimary: "#c6dadf",
      navPrimary: "#191e28",
      buttonPrimary: "#090d19",
    },
  },
  dark: {
    colors: {
      primary: "#fff",
      text: "#fff",
      bgPrimary: "#131b32",
      navPrimary: "#191e28",
      buttonPrimary: "#090d19",
    },
  },
};

describe("<Card />", () => {
  let component;
  beforeEach(() => {
    component = render(
      <Router>
        <ThemeProvider theme={themes.dark}>
          <Card character={character} checkFavorite={() => {}} />
        </ThemeProvider>
      </Router>
    );
  });
  test("renders Card component correctly", () => {
    const component = render(
      <Router>
        <ThemeProvider theme={themes.dark}>
          <Card character={character} checkFavorite={() => {}} />
        </ThemeProvider>
      </Router>
    );
    expect(component.container).toHaveTextContent(character.name);
    expect(component.container).toMatchSnapshot();
  });
});
