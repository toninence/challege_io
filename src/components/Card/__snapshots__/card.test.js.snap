// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> renders Card component correctly 1`] = `
<div>
  <div
    class="sc-hKMtZM dyWGJP carta-box"
  >
    <div
      class="sc-eCYdqJ gQoXJc carta"
    >
      <div
        class="sc-jSMfEi dDKeIo cara"
      >
        <img
          alt="image of Rick Sanchez"
          class="sc-iBkjds hZRQKk"
          src="https://rickandmortyapi.com/api/character/avatar/1.jpeg"
        />
      </div>
      <div
        class="sc-gKXOVf UkGoI cara detras"
      >
        <h3
          class="sc-ftvSup izWOfp"
        >
          Rick Sanchez
        </h3>
        <p
          class="sc-papXJ jtpQEk"
        >
          <svg
            class="sc-dkzDqf dENapR"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 256C0 114.6 114.6 0 256 0C397.4 0 512 114.6 512 256C512 397.4 397.4 512 256 512C114.6 512 0 397.4 0 256zM176.4 240C194 240 208.4 225.7 208.4 208C208.4 190.3 194 176 176.4 176C158.7 176 144.4 190.3 144.4 208C144.4 225.7 158.7 240 176.4 240zM336.4 176C318.7 176 304.4 190.3 304.4 208C304.4 225.7 318.7 240 336.4 240C354 240 368.4 225.7 368.4 208C368.4 190.3 354 176 336.4 176zM160 336C151.2 336 144 343.2 144 352C144 360.8 151.2 368 160 368H352C360.8 368 368 360.8 368 352C368 343.2 360.8 336 352 336H160z"
            />
          </svg>
          Alive
           - 
          Human
        </p>
        <div>
          <span>
            From: 
          </span>
          <p
            class="sc-kDDrLX bmioTm"
          >
             
            Earth (C-137)
          </p>
        </div>
        <div>
          <span>
            Last known location: 
          </span>
          <a
            class="navlink"
            href="/location/3"
          >
            <p
              class="sc-iqcoie cmpjzO"
            >
              Citadel of Ricks
            </p>
          </a>
        </div>
        <div
          style="display: flex; justify-content: space-between;"
        >
          <button
            class="sc-gsnTZi gkpCAO"
          >
            <p>
              Agregar a Favoritos
            </p>
            <svg
              class="sc-bczRLJ dpVPwQ"
              viewBox="0 0 640 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M528 448H112c-8.8 0-16 7.2-16 16v32c0 8.8 7.2 16 16 16h416c8.8 0 16-7.2 16-16v-32c0-8.8-7.2-16-16-16zm64-320c-26.5 0-48 21.5-48 48 0 7.1 1.6 13.7 4.4 19.8L476 239.2c-15.4 9.2-35.3 4-44.2-11.6L350.3 85C361 76.2 368 63 368 48c0-26.5-21.5-48-48-48s-48 21.5-48 48c0 15 7 28.2 17.7 37l-81.5 142.6c-8.9 15.6-28.9 20.8-44.2 11.6l-72.3-43.4c2.7-6 4.4-12.7 4.4-19.8 0-26.5-21.5-48-48-48S0 149.5 0 176s21.5 48 48 48c2.6 0 5.2-.4 7.7-.8L128 416h384l72.3-192.8c2.5.4 5.1.8 7.7.8 26.5 0 48-21.5 48-48s-21.5-48-48-48z"
              />
            </svg>
          </button>
          <a
            class=""
            href="/character/1"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
